ez_setup.py

```diff
@@ -409,5 +410,12 @@ def main():
     return _install(archive, _build_install_args(options))
 
 if __name__ == '__main__':
-    warnings.warn("ez_setup.py is deprecated, check https://github.com/pypa/setuptools/issues/581 for more info; use pip to install setuptools")
+    warnings.warn(DEFAULT_DEPRECATION_MSG)
     sys.exit(main())
+
+# There are situations where you don't use 'python ez_setup.py' but ez_setup.py
+# is still executed (specially when using bootstrap.py from zc.buildout, check
+# https://github.com/buildout/buildout/blob/9ba1e3170a474e6139c9a7eadd32a498c22cb3bb/bootstrap/bootstrap.py#L88
+# for more details). We still need to show the deprecation message in that
+# context.
+log.warn(DEFAULT_DEPRECATION_MSG)
```

Comment:
Won't this cause the warning to be logged twice, once at exec and once when `__name__ == '__main__'`? Perhaps the warning should happen exactly once and closer to the top of the file?